Name: Device Menu Remix
Version: 2.0.1-1
Author: Jason Robitaille
Description: Removes the date, undims the battery level, adds brightness controll, adds GPS toggle, and removes the airplane mode toggle, on the device menu  (MIT license)

--- .orig/usr/lib/luna/system/luna-systemui/app/controllers/devicemenu-assistant.js
+++ /usr/lib/luna/system/luna-systemui/app/controllers/devicemenu-assistant.js
@@ -28,8 +28,6 @@
 		//Add a touchable row class
 		this.controller.get('palm-device-menu').addClassName('palm-touch-rows-'+Mojo.Environment.DeviceInfo.touchableRows);
 			
-		var date = new Date();
-		this.controller.get('dm_date').innerHTML = Mojo.Format.formatDate(date,{date:'EEE '}) +  Mojo.Format.formatDate(date,{date:'long'});
 		var msg = $L("Battery: #{batlevel}");
 		this.controller.get('dm_battery').innerHTML = new Template(msg).evaluate({"batlevel": Mojo.Format.formatPercent(this.barAssistant.getBatteryLevel())});
 						
@@ -32,6 +30,7 @@
 		this.controller.get('dm_date').innerHTML = Mojo.Format.formatDate(date,{date:'EEE '}) +  Mojo.Format.formatDate(date,{date:'long'});
 		var msg = $L("Battery: #{batlevel}");
 		this.controller.get('dm_battery').innerHTML = new Template(msg).evaluate({"batlevel": Mojo.Format.formatPercent(this.barAssistant.getBatteryLevel())});
+		this.controller.setupWidget('brightSlider', {minValue: 1, maxValue: 100}, this.brightModel = { value: 50});
 						
 		this.modesModel = {scrollbars: false, mode: "vertical"},
 		this.controller.setupWidget('devicescroller', {}, this.modesModel);
@@ -38,6 +37,7 @@
 		this.menuscroller = this.controller.get('devicescroller');
 		
 		this.drawerModel = {myOpenProperty:false};
+		this.controller.setupWidget('brightdetails', {modelProperty:'myOpenProperty'}, this.drawerModel);
 		this.controller.setupWidget('imedetails', {modelProperty:'myOpenProperty'}, this.drawerModel);
 		this.controller.setupWidget('wifidetails', {modelProperty:'myOpenProperty'}, this.drawerModel);
 		this.controller.setupWidget('btdetails', {modelProperty:'myOpenProperty'}, this.drawerModel);
@@ -91,10 +91,6 @@
 		//Spinner for BT
 		this.controller.setupWidget('btSpinner', this.spinnerAttrs, this.btspinnerModel);
 			
-		if(this.barAssistant.getAirplaneMode()) 			
-			this.controller.get('dm_airplanemode_status').innerHTML = $L('Turn off Airplane Mode');
-		else
-			this.controller.get('dm_airplanemode_status').innerHTML = $L('Turn on Airplane Mode');
 		
 		var btstate = this.barAssistant.getCurrentBluetoothState();
 		if(btstate === 'Off') {
@@ -147,6 +143,12 @@
 				setText: this.altCharSelected.bind(this)
 			};
 		}
+		this.controller.get("dm_vpn").style.display = "none";
+		var elem = this.controller.get("dm_vpn").previousSibling;
+		while(elem.innerHTML==null || elem.className.indexOf("palm-section-divider")<0)
+			elem = elem.previousSibling;
+		elem.style.display = "none";
+		this.controller.get("vpndetails").style.display = "none";
 	},
 	
 	activate: function() {
@@ -150,7 +152,9 @@
 	},
 	
 	activate: function() {
-		this.controller.get('dm_airplanemode').addEventListener(Mojo.Event.tap, this.toggleAirplaneMode.bind(this));
+		this.controller.get('brightSlider').addEventListener(Mojo.Event.propertyChange, this.handleBrightSlider.bindAsEventListener(this));
+		this.controller.get('brightRow').addEventListener(Mojo.Event.tap, this.handleBrightTap.bindAsEventListener(this));
+		this.controller.get('dm_gps').addEventListener(Mojo.Event.tap, this.toggleGPS.bindAsEventListener(this));
 		this.controller.get('wifilist').addEventListener(Mojo.Event.listTap,this.handleWiFiTap.bind(this));
 		this.controller.get('btlist').addEventListener(Mojo.Event.listTap,this.handleBTTap.bind(this));
 		this.controller.get('vpnlist').addEventListener(Mojo.Event.listTap,this.handleVPNTap.bind(this));		
@@ -175,6 +179,37 @@
 	cleanup: function() {
 		Mojo.Log.error("Device Menu - Clean up called");
 	},	
+	
+	handleBrightTap: function(event) {
+		this.controller.get('brightdetails').mojo.setOpenState(!this.controller.get('brightdetails').mojo.getOpenState());
+	},
+
+	handleBrightSlider: function(event) {
+		this.controller.serviceRequest('palm://com.palm.display/control', {
+			method: 'setProperty',
+			parameters:{maximumBrightness:Math.round(event.value)}
+		});
+		this.controller.get('dm_brightness').innerText = "Brightness: " + Math.round(event.value) + "%";
+	},
+	
+	toggleGPS: function(event) {
+		this.gpsEnabled = !this.gpsEnabled;
+		this.controller.serviceRequest('palm://com.palm.location', {
+			method: 'setUseGps',
+			parameters: {
+				useGps: this.gpsEnabled
+			}
+		});
+		this.setGPSLabel();
+	},
+
+	setGPSLabel: function() {
+		if(this.gpsEnabled) {
+			this.controller.get('gpsStatus').innerText = "On";
+		} else {
+			this.controller.get('gpsStatus').innerText = "Off";
+		}
+	},
 			
 	wifiAPList: [],
 	btdeviceList: [],
@@ -200,7 +235,7 @@
 		if(this.barAssistant.getAirplaneMode()) {
 			Mojo.Log.info("SystemUI - Turning off Airplane Mode - Device Menu");
 			this.barAssistant.setAirplaneMode(false);
-			this.controller.get('dm_airplanemode_status').innerHTML = $L('Turning off Airplane Mode');			
+			//this.controller.get('dm_airplanemode_status').innerHTML = $L('Turning off Airplane Mode');			
 		}			
 		else {
 			Mojo.Log.info("SystemUI - Turning on Airplane Mode - Device Menu");
@@ -205,7 +240,7 @@
 		else {
 			Mojo.Log.info("SystemUI - Turning on Airplane Mode - Device Menu");
 			this.barAssistant.setAirplaneMode(true);
-			this.controller.get('dm_airplanemode_status').innerHTML = $L('Turning on Airplane Mode');
+			//this.controller.get('dm_airplanemode_status').innerHTML = $L('Turning on Airplane Mode');
 		}
 		this.apModeInProgress = true;
 		this.toggleDeviceMenu.delay(0.2);		
@@ -216,12 +251,6 @@
 		//this.apSpinnerModel.spinning = false;
 		//this.controller.modelChanged(this.apSpinnerModel);
 		
-		if(this.barAssistant.getAirplaneMode()) {
-			this.controller.get('dm_airplanemode_status').innerHTML = $L('Turn off Airplane Mode');			
-		}			
-		else {			
-			this.controller.get('dm_airplanemode_status').innerHTML = $L('Turn on Airplane Mode');
-		}
 		this.apModeInProgress = false;
 	},
 	
@@ -227,7 +256,7 @@
 	
 	updateAirplaneModeProgress: function() {
 		if(this.barAssistant.getAirplaneMode()) {
-			this.controller.get('dm_airplanemode_status').innerHTML = $L('Turning on Airplane Mode');
+			//this.controller.get('dm_airplanemode_status').innerHTML = $L('Turning on Airplane Mode');
 		}			
 		else {
 			this.controller.get('dm_airplanemode_status').innerHTML = $L('Turning off Airplane Mode');
@@ -230,7 +259,7 @@
 			this.controller.get('dm_airplanemode_status').innerHTML = $L('Turning on Airplane Mode');
 		}			
 		else {
-			this.controller.get('dm_airplanemode_status').innerHTML = $L('Turning off Airplane Mode');
+			//this.controller.get('dm_airplanemode_status').innerHTML = $L('Turning off Airplane Mode');
 		}
 		this.apModeInProgress = true;
 	},
@@ -1048,7 +1077,7 @@
 	},
 	
 	showVPN:function(available) {
-		if(available) {
+		if(false) {
 			this.controller.get('dm_vpn').show();
 			this.controller.showWidgetContainer(this.controller.get('dm_vpn'));
 		}
@@ -1236,6 +1265,7 @@
 			this.controller.hideWidgetContainer(this.controller.get('wifidetails'));
 			this.wifidrawer.mojo.setOpenState(false);
 		}
+		this.controller.get('brightdetails').mojo.setOpenState(false);
 		if(this.vpndrawer.mojo.getOpenState()) {
 			this.controller.hideWidgetContainer(this.controller.get('vpndetails'));
 			this.vpndrawer.mojo.setOpenState(false);
@@ -1435,8 +1465,27 @@
 		}
 		else {
 			this.isVisible = true;
-			var date = new Date();		
-			this.controller.get('dm_date').innerHTML = Mojo.Format.formatDate(date,{date:'EEE '})  +  Mojo.Format.formatDate(date,{date:'long'});
+			this.controller.serviceRequest('palm://com.palm.display/control', {
+				method: 'getProperty',
+				parameters:{properties:['maximumBrightness']},
+				onSuccess: function(response) {
+					if(response.maximumBrightness != undefined) {
+						this.brightModel.value = parseInt(response.maximumBrightness);
+						this.controller.modelChanged(this.brightModel);
+						this.controller.get('dm_brightness').innerText = "Brightness: " + response.maximumBrightness + "%";
+					} else {
+						this.controller.get('dm_brightness').innerText = "Brightness: ??%";
+					}
+				}.bind(this)
+			});
+			this.controller.serviceRequest('palm://com.palm.location', {
+				method: 'getUseGps',
+				parameters: {},
+				onSuccess: function(response) {
+					this.gpsEnabled = response.useGps;
+					this.setGPSLabel();
+				}.bind(this)
+			});
 			
 			if(this.turningOnBtRadio) {
 				this.btspinnerModel.spinning = true;
--- .orig/usr/lib/luna/system/luna-systemui/app/views/devicemenu/devicemenu-scene.html
+++ /usr/lib/luna/system/luna-systemui/app/views/devicemenu/devicemenu-scene.html
@@ -17,10 +17,12 @@
-				<div class="palm-row first"><div class="palm-row-wrapper">
-					<div id="dm_date" class="title dim"></div>
-				</div></div>
-				
-				<div class="palm-section-divider">
-				</div> 
-				
-				<div class="palm-row"><div class="palm-row-wrapper">
-					<div id="dm_battery" class="title dim"></div>
-				</div></div> 
+					<div class="palm-row first"><div class="palm-row-wrapper">
+				 		<div id="dm_battery" class="title"></div>
+				 	</div></div>
+ 				 	<div class="palm-section-divider"></div>
+				 	<div class="palm-row" id="brightRow" x-mojo-tap-highlight='momentary'><div class="palm-row-wrapper">
+						<div id="dm_brightness" class="title"></div>
+				 	</div></div>
+				 	<div id="brightdetails" x-mojo-element="Drawer">
+					 	<div class="palm-row first"><div class="palm-row-wrapper">
+						 	<div x-mojo-element="Slider" id="brightSlider" style="padding-top: 30px; height: 50px;"></div>
+					 	</div></div>
+				 	</div>
@@ -158,9 +160,11 @@
 				</div>	
 				
 				<div class="palm-section-divider"></div>
-				
-				<div id="dm_airplanemode" class="palm-row last menu-end" x-mojo-tap-highlight='momentary'><div class="palm-row-wrapper">
-					<div id="dm_airplanemode_status" class="title truncating-text">
+
+				 <div class="palm-row last menu-end" id="dm_gps" x-mojo-tap-highlight='momentary'><div class="palm-row-wrapper">
+					 <div class="title truncating-text">
+						 <div class="label right" id="gpsStatus">&nbsp;</div>
+						 <span x-mojo-loc=''>GPS</span>
 					 </div>
 				</div></div>   
 				
